test/generation.test.js

```diff
@@ -25,7 +27,42 @@ describe('Generation tests', () => {
     });
 
     test('Generates headers', () => {
-        expect(headerGenerator.getHeaders()).toBeTruthy();
+        const headers = headerGenerator.getHeaders();
+
+        let userAgent;
+        for (const [header, value] of Object.entries(headers)) {
+            if (header.toLowerCase() === 'user-agent') {
+                userAgent = value;
+                break;
+            }
+        }
+
+        let browser;
+        if (userAgent.includes('Firefox')) {
+            browser = 'firefox';
+        } else if (userAgent.includes('Chrome')) {
+            browser = 'chrome';
+        } else {
+            browser = 'safari';
+        }
+
+        expect(typeof headers).toBe('object');
+
+        const order = headersOrder[browser];
+
+        let index = -1;
+        for (const header of Object.keys(headers)) {
+            const newIndex = order.indexOf(header);
+
+            const log = `${userAgent}\n${browser}\n${inspect(order)}\n${inspect(headers)}`;
+            if (newIndex === -1) {
+                throw new Error(`Missing entry in order array\n${log}`);
+            } else if (newIndex < index) {
+                throw new Error(`Header ${header} out of order\n${log}`);
+            }
+
+            index = newIndex;
+        }
```

Comment:
I don't know if this is a good metric of readability, but I'm trying to understand this while listening to the presentation and I just can't crack it 😄

Reply:
either way, i would vote for not using `else` branch, just like with early returns, it does not make much sense and adds cognitive complexity

Reply:
Ok, fair enough, got it now 😄 Was not that hard after all 😄

Reply:
See comments below :) Please ask any questions if any :)

Reply:
No more questions :) I think it's ok. I would just prefer if you refactored the prepwork into functions with names so that it's absolutely clear what the tested units are and what's just preprocessing of data and so on. And also use `expect` rather than plain `throw` to make it clear again what the tested values are.

Reply:
I mean, the logic is good. It just does not look like a test :)

Reply:
But then we would be missing crucial data and it would be harder to debug. This way if it throws we know what the headers are, what's the user-agent, what's the

Reply:
Hmm, you're right. My bad. I thought you can add custom messages to `expect` quite easily, but I was wrong. Then please just add a comment above the errors that explain the expectations as you explained in the PR comments 👍